src/components/common/ImageContainer/index.tsx

```diff
@@ -0,0 +1,44 @@
+import Image, { ImageProps } from 'next/image';
+
+type ImageType = 'profile' | 'small' | 'medium' | 'full';
+
+interface ImageContainerProps {
+  type: ImageType;
+  src: string;
+}
+
+interface ImageAttributes extends Pick<ImageProps, 'width' | 'height'> {
+  radius: string;
+}
+
+const getImageAttributes = (type: ImageType): ImageAttributes => {
+  switch (type) {
+    case 'profile':
+      return { width: 100, height: 100, radius: 'rounded-full' };
+    case 'small':
+      return { width: 100, height: 100, radius: 'rounded-[20px]' };
+    case 'medium':
+      return { width: 343, height: 150, radius: 'rounded-[20px]' };
+    case 'full':
+      return { width: 375, height: 280, radius: 'rounded-none' };
+  }
+};
+
+export default function ImageContainer({
+  type,
+  src,
+  ...restProps
+}: ImageContainerProps) {
+  const { width, height, radius } = getImageAttributes(type);
+
+  return (
+    <Image
+      src={src}
+      width={width}
+      height={height}
+      alt="image"
+      className={radius}
+      {...restProps}
+    />
+  );
+}
```

Comment:
next Image에서 사용되는 prop을 모두 사용하기 위해 아래처럼 수정하는건 어떨까요??

```suggestion
export default function ImageContainer({type, className, ...restProps}: ImageContainerProps) {
  const { width, height, radius } = getImageAttributes(type);

  return (
    <Image
      width={width}
      height={height}
      className={cn(radius, className)}
      {...restProps}
    />
  );
}
```